sites/main-site/src/content/blog/2024/seqera-containers-part-1.mdx

```diff
@@ -0,0 +1,360 @@
+    </tbody>
+</table>
+
+<small>
+    \* Forever is a long time. Seqera is committing to keeping images for a minimum of 5 years from the time of their
+    creation. BioContainers has no public policy.
+</small>
+
+:::
+
+### Mulled (multi-package) images
+
+<Image
+    class="d-block ms-3 mb-3 float-end rounded w-25 shadow"
+    src={mulled}
+    alt="Photo by Hannah Pemberton on Unsplash."
+/>
+
+BioContainers is primarily set up to have a 1:1 relationship with Bioconda. This is great for bioinformatics tools: When a package is published, it gets an image. This works well until you need to use more than one tool in a single process, and particularly when the tools are from elsewhere in the conda ecosystem. For example, many bioinformatics tools leverage [samtools](https://www.htslib.org/) to convert file formats or sort reads on the fly. Others may pipe output to compression tools like [pigz](https://zlib.net/pigz/). To resolve this, BioContainers has the concept of “mulled” images (as in [mulled wine](https://en.wikipedia.org/wiki/Mulled_wine)).
+
+Unfortunately, generating mulled containers is not trivial. Luke Pembleton, Nextflow Ambassador, has a great blog post summarising the dark art of [Finding the right mulled biocontainer](https://lpembleton.rbind.io/posts/mulled-biocontainers/). Galaxy also has [documentation on the topic](https://docs.galaxyproject.org/en/master/admin/container_resolvers.html). In short, to request an image you edit a [CSV file of hashes](https://github.com/BioContainers/multi-package-containers/blob/master/combinations/hash.tsv) in a GitHub repo, scroll all the way to the bottom, and add a line with your requested conda packages. Once edited, the images are built on CI. It used to be that you had to hunt through the GitHub action log to find the URI for your new container. However, [Moriz E. Beber](https://github.com/Midnighter) from the nf-core community has created a webpage to [generate the name of the mulled container](https://midnighter.github.io/mulled) which gently guides you through the process of finding the name of your container and how to update a container if you want to bump the software versions. Once the image is built and you know its address, the final step is to go to nf-core/modules and create a pull-request with the updated containers and bump the versions in the conda `environment.yml`.
+
+This system works and, despite its complexity, is now a familiar process to many nf-core developers. However it does present some problems. It's a highly manual process to update software versions and update the container declarations – just fetching the software for a module often ends up taking longer than writing the entire module. There are also significant delays in waiting for the images to become available. All in all, it can be a frustrating experience and we see this in the volume of Slack messages asking for help. It seems likely that this puts some people off from contributing to nf-core.
```

Comment:
```suggestion
This system works and, despite its complexity, is now a familiar process to many nf-core developers. However, it does present some problems. It's a highly manual process to update software versions and the container declarations – just fetching the software for a module often ends up taking longer than writing the entire module. There are also significant delays in waiting for the images to become available. All in all, it can be a frustrating experience and we see this in the volume of Slack messages asking for help. It seems likely that this puts some people off from contributing to nf-core.
```